### PR TITLE
moved MEMOIZE_LIMIT to kwarg of function it's used in

### LINKED ISSUE #2884: Value from outside scope of function referenced in docstring
https://github.com/cylc/cylc/blob/daf7dee07f44b9c105453f865b8cef16d920af27/lib/cylc/cycling/iso8601.py#L65

Minor thing, but I feel that we shouldn't reference the value contained in this variable in the docstring and should either:
* Set `MEMOIZE_LIMIT` inside the function.
* Pass `MEMOIZE_LIMIT` as a function kwarg (seems like a plan to me...).

### PR DESCRIPTION
**Pull request by wxtim**
Fix #2884

### CODE REVIEW

File: lib/cylc/cycling/iso8601.py [hunk 1]
```diff
@@ -56,7 +54,7 @@ class SuiteSpecifics(object):
     iso8601_parsers = None
 
 
-def memoize(function):
+def memoize(function, MEMOIZE_LIMIT=10000):
```
Comment (wxtim) [inline lib/cylc/cycling/iso8601.py]: Should I use Google docstring style?
```python
Args:
    function (function):
        Lorum Ipsum....
Kwargs:
    MEMOIZE_LIMIT (int):
        Stuff
```

File: lib/cylc/cycling/iso8601.py [hunk 2]
```diff
@@ -56,15 +54,18 @@ class SuiteSpecifics(object):
     iso8601_parsers = None
 
 
-def memoize(function):
-    """This stores results for a given set of inputs to a function.
-
-    The inputs and results of the function must be immutable.
-    Keyword arguments are not allowed.
+def memoize(function, MEMOIZE_LIMIT=10000):
```
Comment (kinow) [inline lib/cylc/cycling/iso8601.py]: I think it can be `memoize_limit` now, as it's not a constant anymore?

Comment (hjoliver) [reply lib/cylc/cycling/iso8601.py]: Agreed, if kept as a function arg ... but I tend to think it should remain as a constant (either at module level, or internal to the `memoize` function).  This is used as a decorator, to modify the behavior of a bunch of methods, and do we have any reason to use different cache sizes in different methods at this stage??

Comment (hjoliver) [reply lib/cylc/cycling/iso8601.py]: (Obviously that could be useful in principle, but at the moment we're only using the 10k size)

Comment (kinow) [reply lib/cylc/cycling/iso8601.py]: (And once py3, we can look into functools lru cache too)

Comment (wxtim) [reply lib/cylc/cycling/iso8601.py]: I have moved it to inside the outer function. I could have put the number directly into the line `if len(inputs_results) > memoize_limit:`, if the value is to be stored inside this function, but I think setting it makes the fn more readable. 

It isn't used anywhere else so I don't think it should be global in scope.

No-one wants to set it from outside at the moment, so it doesn't need to be a kwarg.

**Comment by wxtim** [comment]
Closed. Not worth the effort.

**Pull request closed without merging**